provider/datagen/src/transform/segmenter/dictionary.rs

```diff
@@ -65,19 +47,22 @@ impl crate::DatagenProvider {
 
         let filename = format!("segmenter/dictionary/{model}.toml");
 
-        let toml_data: &SegmenterDictionaryData = self
-            .source
+        let toml_data = self
             .icuexport()
-            .and_then(|e| e.read_and_parse_toml(&filename))
-            .or_else(|e| {
-                self.source
-                    .icuexport_fallback()
-                    .read_and_parse_toml(&filename)
-                    .map_err(|_| e)
-            })?;
+            .and_then(|e| e.read_and_parse_toml::<SegmenterDictionaryData>(&filename));
+
+        #[cfg(feature = "legacy_api")]
```

Comment:
question: is it new in this PR that you're gating the segmenter data on the legacy_api feature? Might be something worth pulling out for further discussion but I'm okay landing it

Reply:
Yes this is new, now that we have `DatagenProvider` and `SourceData`, I think it's best to require clients to set it on `DatagenProvider`. Built-in data is a bit of a footgun.

Reply:
ok